Makefile

```diff
@@ -244,35 +203,8 @@ flash: build/$(ESP32_CHIP)/flash/program.snapshot
 endif
 
 .PHONY: flash
-flash: check-env-flash sdk esp32
-	mkdir -p build/$(ESP32_CHIP)/flash
-	cp build/$(ESP32_CHIP)/firmware.envelope build/$(ESP32_CHIP)/flash/firmware.envelope
-ifdef ESP32_ENTRY
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    container install program build/$(ESP32_CHIP)/flash/program.snapshot
-endif
-ifdef ESP32_WIFI_SSID
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    property set wifi '{"wifi.ssid": "$(ESP32_WIFI_SSID)", "wifi.password": "$(ESP32_WIFI_PASSWORD)"}'
-endif
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    extract --bootloader.bin -o build/$(ESP32_CHIP)/flash/bootloader.bin
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    extract --partitions.bin -o build/$(ESP32_CHIP)/flash/partitions.bin
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    extract --firmware.bin -o build/$(ESP32_CHIP)/flash/firmware.bin
-	python $(IDF_PATH)/components/esptool_py/esptool/esptool.py --chip $(ESP32_CHIP) --port $(ESP32_PORT) --baud 921600 \
-	    --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 40m --flash_size detect \
-		0x001000 build/$(ESP32_CHIP)/flash/bootloader.bin \
-		0x008000 build/$(ESP32_CHIP)/flash/partitions.bin \
-		0x010000 build/$(ESP32_CHIP)/flash/firmware.bin
-
-.PHONY: check-env-flash
-check-env-flash:
-ifndef ESP32_PORT
-	$(error ESP32_PORT is not set)
-endif
-
+flash: check-env check-esp32-env sdk toolchains/$(ESP32_CHIP)/sdkconfig
+	(cd toolchains/$(ESP32_CHIP) && idf.py -B ../../build/$(ESP32_CHIP) -p $(ESP32_PORT) flash monitor)
```

Comment:
This will fail in a non-obvious way if `ESP32_PORT` is not set.
It should either check for it or not need it (or have a default value).

Reply:
Yes, I added "" around $(ESP32_PORT). Ideally it would be better to give an error.